binrw_derive/src/parser/top_level_attrs.rs

```diff
@@ -128,9 +134,13 @@ impl Struct {
     }
 
     pub(crate) fn iter_permanent_idents(&self) -> impl Iterator<Item = &syn::Ident> + '_ {
-        self.fields
-            .iter()
-            .filter_map(|field| if field.temp { None } else { Some(&field.ident) })
+        self.fields.iter().filter_map(|field| {
+            if field.temp.is_some() {
+                None
+            } else {
+                Some(&field.ident)
+            }
```

Comment:
Maybe this?
```suggestion
            field.temp.is_none().then(|| &field.ident)
```

Reply:
Yeah, I blindly added `is_none` & `is_some` everywhere, `bool::then` is more appropriate here.